tests/processing/test_create_plate_processor.py

```diff
@@ -112,23 +122,103 @@ def test_process_when_transient_error(subject, mock_logger, mock_validator):
     assert ex_info.value == transient_error
 
 
-def test_process_when_another_exception(subject, mock_logger, mock_validator):
+def test_process_when_transient_error_from_exporter(subject, mock_logger, mock_exporter):
+    transient_error = TransientRabbitError("Test transient error")
+    mock_exporter.return_value.export_to_mongo.side_effect = transient_error
+
+    with pytest.raises(TransientRabbitError) as ex_info:
+        subject.process(MagicMock())
+
+    mock_logger.error.assert_called_once()
+    assert ex_info.value == transient_error
+
+
+def test_process_when_another_exception_from_the_validator(subject, mock_logger, mock_validator, message_wrapper_class):
     another_exception = KeyError("key")
     mock_validator.return_value.validate.side_effect = another_exception
-    with patch("crawler.processing.create_plate_processor.CreatePlateMessage") as create_plate_message:
-        with patch(
-            "crawler.processing.create_plate_processor.CreatePlateProcessor._publish_feedback"
-        ) as publish_feedback:
-            result = subject.process(MagicMock())
+    with patch("crawler.processing.create_plate_processor.CreatePlateProcessor._publish_feedback") as publish_feedback:
```

Comment:
We are patching a method that is suppose to be private. There is a risk on future refactors not complying with it when there is a change in the implementation of the public methods that use this private method.
Maybe it means that this method should be recognised as part of the official API, should this method _publish_feedback() be public?

Reply:
No it shouldn't be public.  It is called by the method we are testing only.  Nothing outside the CreatePlateProcessor should be calling it.  It's only being patched so that we can assert that it was called by the public method.  However, we can, if desired, assert that the public method that called `_publish_feedback` is doing the things that `_publish_feedback` does.  It's just that when you do that, you end up repeating test assertions over and over if more than one public method calls the same private method.  I'm happy to change it to that though.  I think you're right.